src/main/java/uk/gov/hmcts/darts/datamanagement/service/impl/DataManagementServiceImpl.java

```diff
@@ -63,6 +66,26 @@ public BinaryData getBlobData(String containerName, UUID blobId) {
         return binaryData;
     }
 
+    @Override
+    @SneakyThrows
+    public Path downloadBlobToFile(String containerName, UUID blobId, String workspace) {
+        BlobServiceClient serviceClient = blobServiceFactory.getBlobServiceClient(dataManagementConfiguration.getBlobStorageAccountConnectionString());
+        BlobContainerClient containerClient = blobServiceFactory.getBlobContainerClient(containerName, serviceClient);
+        BlobClient blobClient = blobServiceFactory.getBlobClient(containerClient, blobId);
+        if (!blobClient.exists()) {
+            log.error("Blob {} does not exist in {} container", blobId, containerName);
+        }
+
+        Path workspacePath = Path.of(workspace);
+        Path targetFile = workspacePath.resolve(UUID.randomUUID() + ".tmp");
```

Comment:
Can this change the current checksum of the file? If so, this will cause issues in further validation.

Reply:
the checksum does not depend on the file name, only on the content so I don't think. Verified locally with Mac Os `md5` command line tool